docs/metrics/perplexity.md

```diff
@@ -0,0 +1,118 @@
+---
+search:
+  exclude: true
+---
+
+# Perplexity
+
+!!! info inline end "Requirements"
+
+    It is important to note that perplexity can only be calculated using autoregressive language models.
+
+Perplexity is a metric commonly used in natural language processing to evaluate the quality of language models, particularly in the context of text generation. Unlike metrics such as BLEU or BERT, perplexity doesn't directly measure the quality of generated text by comparing it with reference texts. Instead, perplexity assesses the "confidence" or "surprise" of a language model in predicting the next word in a sequence of words.
+
+Perplexity quantifies how well a language model can predict the next word in a sequence of words. It is calculated based on the probability distribution of words generated by the model. A high perplexity indicates that the language model is not confident in its text generation — that is, the model is "perplexed" — whereas a low perplexity indicates that the language model is confident in its generation.
+
+??? question "Autoregressive Language Models"
```

Comment:
Suggestion: remove the top callout and replace this with:

```
??? note "Note: Perplexity can only be calculated for autoregressive language models"
```

![Screenshot 2023-10-20 at 10 40 35 AM](https://github.com/kolenaIO/kolena/assets/9648886/70701cd8-bb3b-41e4-a101-d14bd7729cfa)